### PR TITLE
SSL: "Bring your own CA" support & generate SSL locally

### PR DESCRIPTION
**Pull request by EmilienM**
# Bring your own CA
Allow a user to bring their own Certificate Authority to dev-install,
so when OpenStack will be deployed, the generated certificate will be
signed with this CA.

It's a famous customer use-case, and will also be used by our CI, so we
can rely on a stable CA and just generate ephemeral certificates when
deploying OpenStack.

# Generate SSL files locally instead of remotely

Instead of generating all SSL files remotely, do it locally, it's more
secure.

The main reason of doing this is because we do not want the CA private
key to be on the remote host, this would be a security issue in case
someone has access to the host, they can compromise our CA

### CODE REVIEW

**Comment by EmilienM** [comment]
I'm still testing it now and I think I found an issue, please don't merge it yet.

**Comment by EmilienM** [comment]
tested in PSI and it worked fine.

File: playbooks/install_stack.yaml
Comment (mandre) [inline playbooks/install_stack.yaml:56]: No need, the parent task already has `when: ssl_enabled`

File: playbooks/install_stack.yaml
Comment (mandre) [inline playbooks/install_stack.yaml:82]: Should be `ssl_key`, I'll address the issue in a separate patch.

**Review by mandre — COMMENTED** [review]
Happy to merge this once you had a chance to look at my comments.

File: playbooks/install_stack.yaml
Comment (mandre) [reply playbooks/install_stack.yaml:82]: I meant to create a PR and accidentally pushed to main, sorry :angel: 

https://github.com/shiftstack/dev-install/commit/6724a1daa3b7834871642781913a66672b92d6b7

File: playbooks/vars/defaults.yaml
Comment (mdbooth) [inline playbooks/vars/defaults.yaml:105]: nit: /and/

File: playbooks/install_stack.yaml
Comment (mdbooth) [reply playbooks/install_stack.yaml:82]: Ah, that will also be why there's a merge conflict.

**Review by mdbooth — COMMENTED** [review]
Need to fix the merge conflict, obviously. Might as well fix the typo while you're at it, but that's not important.

Something to think about: by generating a new CA every time this executes (when you're not passing a pre-created one in), I think this means install_stack can never be idempotent. It's not idempotent right now and I doubt it ever will be, so I suspect that's not a problem in practise.

**Review by mandre — APPROVED** [review]
:+1:

**Pull request merged**
